tensorpipe/common/cuda_loop.cc

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#include <tensorpipe/common/cuda_loop.h>
+
+#include <tensorpipe/common/cuda.h>
+#include <tensorpipe/common/system.h>
+
+namespace tensorpipe {
+
+CudaLoop::CudaLoop() {
+  std::unique_lock<std::mutex> lock(mutex_);
```

Comment:
Nit: no point in acquiring the lock in the constructor: no one else has yet a pointer to this class, so no one can call any method on it.